packages/alfa-rules/src/sia-r28/rule.ts

```diff
@@ -39,9 +39,15 @@ export default Rule.Atomic.of<Page, Element>({
       expectations(target) {
         return {
           1: expectation(
-            hasNonEmptyAccessibleName(device)(target),
-            () => Outcomes.HasName,
-            () => Outcomes.HasNoName
+            test(
+              hasAccessibleName(
+                device,
+                (accessibleName) => accessibleName.value !== "Submit Query"
+              ),
+              target
```

Comment:
Thinking back on this…
We might be able to determine that the name comes from the default value by checking its source: https://github.com/Siteimprove/alfa/blob/f37f76b14fc856f8c76633be8bd21eff91778f9d/packages/alfa-aria/src/name.ts#L47-L49
If I get it properly, default names should have an empty source, so
```suggestion
              hasAccessibleName(
                device,
                (name) => name.sources.length !==0
              ),
              target
```
That way we're solving the problem of duplicating the default name 😄 ; and we accept `<input type="image" aria-label="Submit Query" />` where we can assume that the devs know what they do and that the name is actually good.